### PR TITLE
ojopaul:feature/#24_Allow_authors_to_update_their_own_information

### PR DESCRIPTION
**Pull request by ojopaul**
## Description

- Allow authors to update their own information #24

We introduce 'ppma_edit_own_profile' capability which is added to 'administrator', 'editor' and 'author' role on migration.

This added a new admin menu "Author Profile" for users that match the following criterias:
- Users without ppma_manage_authors capability
- Users with a mapped author profile

In addition, when user is editing own profile, mapped user selection and delete button are hidden with css for usability purpose as without hiding the delete button user that cannot delete author profile without ppma_manage_authors capability

## Benefits

User will now see "Author Profile" menu which allow them to edit their mapped author profile

## Applicable issues

https://github.com/publishpress/PublishPress-Authors/issues/24

## Checklist

- [x] I have created a specific branch for this pull request before committing, starting off the current HEAD of `development` branch. 
- [x] I'm submitting to the `development`, feature/hotfix/release branch. (Do not submit to the master branch!)
- [x] This pull request relates to a specific problem (bug or improvement).
- [x] I have mentioned the issue number in the pull request description text.
- [x] All the issues mentioned in this pull request relate to the problem I'm trying to solve.
- [x] The code I'm sending follows the [PSR-12](https://www.php-fig.org/psr/psr-12/) coding style.

### CODE REVIEW

**Comment by andergmartins** [comment]
I have cloned your repo and tested the code.
Using a user from role "author", I can see the author profile menu, but when I try to edit the profile after clicking on the save button it returns an error: "Item not updated."

image


The "author" role has the following capabilities set:

* [ ] ppma_manage_authors (no)
* [ ] ppma_edit_post_authors (no)
* [X] ppma_edit_own_profile (yes)

**Review by andergmartins — CHANGES_REQUESTED** [review]
@ojopaul Works well, just need to review the notes I added.

File: src/modules/multiple-authors/multiple-authors.php [hunk 1]
```diff
@@ -2068,16 +2093,27 @@ public function admin_enqueue_scripts()
...
+                }
 
-            $legacyOptions = get_option('publishpress_multiple_authors_options');
-            if (!empty($legacyOptions)) {
-                update_option('multiple_authors_multiple_authors_options', $legacyOptions);
+                update_option('publishpress_multiple_authors_settings_migrated_3_0_0', 1);
+            }
+            
+            if (!get_option('publishpress_multiple_authors_settings_migrated_3_15_0')) {
+               if(function_exists('get_role')){
```
[the remark this answers is not in the document]
Comment (ojopaul) [reply src/modules/multiple-authors/multiple-authors.php:2106]: Thanks, will check

**Comment by ojopaul** [comment]
@andergmartins , i'm not able to recreate the "Item not updated issue" after many trial like adding new author and testing directly.

I believe this may not be related to permission issue or this specific PR, can you try to edit this same profile as admin and see if you get same response ?

**Review by andergmartins — APPROVED** [review]

**Comment by andergmartins** [comment]
Thanks @ojopaul it is working great

**Pull request merged**